main.go

```diff
@@ -48,9 +50,38 @@ func arnFilename(arn string) string {
 
 func printCredentials(result *result) error {
 	if *envFormat {
-		fmt.Printf("export AWS_ACCESS_KEY_ID=%s\n", *result.Credentials.AccessKeyId)
-		fmt.Printf("export AWS_SECRET_ACCESS_KEY=%s\n", *result.Credentials.SecretAccessKey)
-		fmt.Printf("export AWS_SESSION_TOKEN=%s\n", *result.Credentials.SessionToken)
+		// Get the name of current user's default shell
+		default_shell := os.Getenv("SHELL")
+
+		current_shell := path.Base(default_shell)
+
+		// If the user has specified a shell, use that instead
+		if *shell != "" {
+			current_shell = *shell
+		}
+
+		// Check the shell type and print the appropriate command to export the variable
+		switch current_shell {
+		case "bash", "zsh", "sh":
+			// For bash, zsh and sh, use the export command
+			fmt.Printf("export AWS_ACCESS_KEY_ID=%s\n", *result.Credentials.AccessKeyId)
+			fmt.Printf("export AWS_SECRET_ACCESS_KEY=%s\n", *result.Credentials.SecretAccessKey)
+			fmt.Printf("export AWS_SESSION_TOKEN=%s\n", *result.Credentials.SessionToken)
+		case "fish":
+			// For fish, use the set command
+			fmt.Printf("set -x AWS_ACCESS_KEY_ID %s\n", *result.Credentials.AccessKeyId)
+			fmt.Printf("set -x AWS_SECRET_ACCESS_KEY %s\n", *result.Credentials.SecretAccessKey)
+			fmt.Printf("set -x AWS_SESSION_TOKEN %s\n", *result.Credentials.SessionToken)
+		case "csh", "tcsh":
+			// For csh and tcsh, use the setenv command
+			fmt.Printf("setenv AWS_ACCESS_KEY_ID %s\n", *result.Credentials.AccessKeyId)
+			fmt.Printf("setenv AWS_SECRET_ACCESS_KEY %s\n", *result.Credentials.SecretAccessKey)
+			fmt.Printf("setenv AWS_SESSION_TOKEN %s\n", *result.Credentials.SessionToken)
+		default:
+			// For other shells, return an error
+			return errors.Errorf("unsupported shell: %s", current_shell)
```

Comment:
This is changing the default behavior. It probably won't affect many people, though. I'll leave it to you.

Reply:
Good point. I'll change this here for consistency with the previous behavior.